File: Project/Engine/Graphics/Buffer/Data/UVTransform.cpp
#include "UVTransform.h"

using namespace ONEngine;


/// ------------------------------------------------
/// UVTransform Json変換
/// ------------------------------------------------

void ONEngine::to_json(nlohmann::json& j, const UVTransform& uvTransform) {
	j = nlohmann::json{
		{ "position", uvTransform.position },
		{ "scale", uvTransform.scale },
		{ "rotate", uvTransform.rotate },
	};
}

void ONEngine::from_json(const nlohmann::json& j, UVTransform& uvTransform) {
	uvTransform.position = j.value("position", Vector2::Zero);
	uvTransform.scale = j.value("scale", Vector2::One);
	uvTransform.rotate = j.value("rotate", 0.0f);
}

File: SubProjects/CSharpLibrary/Scripts/Engine/Graphics/UVTransform.cs
using System.Runtime.InteropServices;

[StructLayout(LayoutKind.Sequential)]
public struct UVTransform {
	public Vector2 position;
	public Vector2 scale;
	public float rotate;
	private float pad1;
	private float pad2;
	private float pad3;

	public static UVTransform identity {
		get {
			UVTransform uv = new UVTransform();
			uv.position = Vector2.zero;
			uv.scale = Vector2.one;
			uv.rotate = 0.0f;
			return uv;
		}
	}
}
